scripts/repoint-latest

```diff
@@ -0,0 +1,175 @@
+# limitations under the License.
+
+set -e
+
+DRYRUN=true
+NUCLEAR=false
+
+IMAGE_NAME="amazon/amazon-ecs-agent"
+
+AWS_PROFILE=""
+PUBLISH_S3_BUCKET="amazon-ecs-agent"
+S3_ACL_OVERRIDE=""
+
+source $(dirname "${0}")/publishing-common.sh
+
+TARGET_OS_LIST=${SUPPORTED_OSES[@]}
+
+usage() {
+	echo "Usage: ${0} -s BUCKET [OPTIONS]"
+	echo
+	echo "This script is responsible for un-publishing a given release of the ECS Agent."
+	echo		
+	echo "Options"
+	echo "  -d  true|false      Dryrun (default is true)"
+	echo "  -b  BAD_VERSION     Version that will be unpublished"
+	echo "  -g  GOOD_VERSION    Version that will be re-tagged as latest"
+	echo "  -p  PROFILE         AWS CLI Profile (default is none)"
+	echo "  -n                  Nuclear option: delete the current version rather than just moving the latest pointer"
+	echo "  -h                  Display this help message"
+}
+
+unpublish_s3() {
+	if [ "$1" == "windows" ]; then
+	     	base_name="ecs-agent-windows"
+		suffix="zip"
+	else
+		base_name="ecs-agent"
+		suffix="tar"
+	fi
+
+	tarball_bad="$(mktemp)"
+	tarball_latest="$(mktemp)"
+
+	s3_cp "s3://${PUBLISH_S3_BUCKET}/ecs-agent-${BAD_VERSION}.tar" "${tarball_bad}"
+	s3_cp "s3://${PUBLISH_S3_BUCKET}/ecs-agent-${IMAGE_TAG_LATEST}.tar.md5" "${tarball_latest}"
+
+	if ! check_md5 "${tarball_bad}" test_md5="$(md5sum ${tarball_latest} | sed 's/ .*//')" ; then
+		echo "The bad version ${BAD_VERSION} is not currently tagged as latest"
+		rm "${tarball_bad}"
+		rm "${tarball_latest}"
+		exit 1
+	fi
+
+	set +e
+	s3_ls "s3://${PUBLISH_S3_BUCKET}/${base_name}-${GOOD_VERSION}.${suffix}" &> /dev/null
+	if [ $? -ne 0 ]; then
+		echo "Unable to find prior version ${base_name}-${GOOD_VERSION}.${suffix} in ${PUBLISH_S3_BUCKET}"
+		exit 1
+	fi
+	set -e
+	
+	echo "Un-Publishing ${base_name}-${BAD_VERSION}.${suffix} as latest"
+	echo "Marking ${base_name}-${GOOD_VERSION}.${suffix} as latest"
+
+	dryval s3_cp "s3://${PUBLISH_S3_BUCKET}/${base_name}-${GOOD_VERSION}.${suffix}" \
+		     "s3://${PUBLISH_S3_BUCKET}/${base_name}-${IMAGE_TAG_LATEST}.${suffix}"
+
+	if ${NUCLEAR}; then
+		dryval s3_rm "s3://${PUBLISH_S3_BUCKET}/${base_name}-${BAD_VERSION}.${suffix}"
+	   	dryval s3_rm "s3://${PUBLISH_S3_BUCKET}/${base_name}-${IMAGE_TAG_SHA}.${suffix}"	   
+	fi
+}
+
+unpublish_docker() {
+	if [ "$1" == "windows" ]; then
+		echo "Skipping docker publication on Windows"
+		return
+	fi
+
+	tarball="$(mktemp)"
+	tarball_md5="$(mktemp)"
+
+	s3_cp "s3://${PUBLISH_S3_BUCKET}/ecs-agent-${GOOD_VERSION}.tar" "${tarball}"
+	s3_cp "s3://${PUBLISH_S3_BUCKET}/ecs-agent-${GOOD_VERSION}.tar.md5" "${tarball_md5}"
+
+	if ! check_md5 "${tarball}" "${tarball_md5}" ; then
+		echo "Failed to validate integrity of s3://${PUBLISH_S3_BUCKET}/ecs-agent-${GOOD_VERSION}.tar"
+		rm "${tarball}"
+		rm "${tarball_md5}"
+		exit 1
+	fi
+
+	docker load < "${tarball}"
+
+	echo "Tagging as ${IMAGE_NAME}:${IMAGE_TAG_LATEST}"
+	docker tag amazon/amazon-ecs-agent:latest "${IMAGE_NAME}:${IMAGE_TAG_LATEST}"
```

Comment:
Should we also have something here to remove all of `amazon/amazon-ecs-agent:*` images from the local file system?